### PR TITLE
fix: review for code smell, duplicates, IDE notices

### PR DESCRIPTION
**Pull request by caseyw**
I noticed that there were some code smells, duplicates, and IDE notices in the codebase.

Pretty normal stuff, but I wanted to clean it up a bit while we were in there.

Let me know if anyone has any questions.

### CODE REVIEW

File: src/ChangeLeadTimeGraph.tsx [hunk 1]
```diff
@@ -35,7 +35,7 @@ interface ProcessRepository {
 }
 
 export const composeGraphData = (_: ChartProps, data: DoraRecord[]) => {
-  let reduced = data.reduce(
+  return data.reduce(
```
Comment (Wolftousen) [inline src/ChangeLeadTimeGraph.tsx]: I'm in general not a fan of doing any sort of logic as part of a return statement.  It only saves 1 line of code but when trying to debug can cause headaches trying to step through things and set breakpoints.

File: src/TrendGraph.tsx [hunk 2]
```diff
@@ -152,11 +152,9 @@ const filterGraphData = (
   start: number,
   end: number,
 ): GraphData[] => {
-  const filtered = data.filter((entry: GraphData) => {
+  return data.filter((entry: GraphData) => {
```
Comment (Wolftousen) [inline src/TrendGraph.tsx]: see previous comment about logic in return statements

File: src/functions/metricFunctions.tsx [hunk 3]
```diff
@@ -118,9 +118,7 @@ const calculateDeploymentFrequencyAverage = (
     totalDeployTime += diff;
   }
 
-  let avgDeployTime = totalDeployTime / sorted.length / millisecondsToHours;
-
-  return avgDeployTime;
+  return totalDeployTime / sorted.length / millisecondsToHours;
```
Comment (Wolftousen) [inline src/functions/metricFunctions.tsx]: see previous comment about logic in return statements

File: stories/Board.stories.tsx [hunk 4]
```diff
@@ -1,7 +1,7 @@
 import React, { useState } from 'react';
 import { StoryFn, Meta } from '@storybook/react';
 import Board from '../src/Boards/Board';
-import { ChartProps, Theme } from '../src/interfaces/propInterfaces';
+import { ChartProps, Theme } from '../src';
```
Comment (Wolftousen) [inline stories/Board.stories.tsx]: i don't think we should be referencing index.ts internally, I'd rather be explicit with internal code on where something is being imported from.  To me index.ts is just for exporting to the consumer of this component set.

File: stories/ChangeFailureRate.stories.tsx [hunk 5]
```diff
@@ -1,7 +1,7 @@
 import React from 'react';
 import { StoryFn, Meta } from '@storybook/react';
 import ChangeFailureRateGraph from '../src/ChangeFailureRateGraph';
-import { ChartProps, Theme } from '../src/interfaces/propInterfaces';
+import { ChartProps, Theme } from '../src';
```
Comment (Wolftousen) [inline stories/ChangeFailureRate.stories.tsx]: see previous comment about index.ts referencing

File: stories/ChangeLeadTime.stories.tsx [hunk 6]
```diff
@@ -1,7 +1,7 @@
 import React from 'react';
 import { StoryFn, Meta } from '@storybook/react';
 import ChangeLeadTimeGraph from '../src/ChangeLeadTimeGraph';
-import { ChartProps, Theme } from '../src/interfaces/propInterfaces';
+import { ChartProps, Theme } from '../src';
```
Comment (Wolftousen) [inline stories/ChangeLeadTime.stories.tsx]: see previous comment about index.ts referencing

File: stories/DeploymentFrequency.stories.tsx [hunk 7]
```diff
@@ -1,7 +1,7 @@
 import React from 'react';
 import { StoryFn, Meta } from '@storybook/react';
 import DeploymentFrequencyGraph from '../src/DeploymentFrequencyGraph';
-import { ChartProps, Theme } from '../src/interfaces/propInterfaces';
+import { ChartProps, Theme } from '../src';
```
Comment (Wolftousen) [inline stories/DeploymentFrequency.stories.tsx]: see previous comment about index.ts referencing

File: stories/MetricEditor.tsx [hunk 8]
```diff
@@ -1,5 +1,5 @@
 import React from 'react';
-import { MetricThresholds } from '../src/interfaces/propInterfaces';
+import { MetricThresholds } from '../src';
```
Comment (Wolftousen) [inline stories/MetricEditor.tsx]: see previous comment about index.ts referencing

File: stories/RecoverTime.stories.tsx [hunk 9]
```diff
@@ -1,7 +1,7 @@
 import React from 'react';
 import { StoryFn, Meta } from '@storybook/react';
 import RecoverTimeGraph from '../src/RecoverTimeGraph';
-import { ChartProps, Theme } from '../src/interfaces/propInterfaces';
+import { ChartProps, Theme } from '../src';
```
Comment (Wolftousen) [inline stories/RecoverTime.stories.tsx]: see previous comment about index.ts referencing

File: stories/Trends.stories.tsx [hunk 10]
```diff
@@ -1,7 +1,7 @@
 import React, { useState } from 'react';
 import { StoryFn, Meta } from '@storybook/react';
 import TrendGraph from '../src/TrendGraph';
-import { ChartProps, Theme } from '../src/interfaces/propInterfaces';
+import { ChartProps, Theme } from '../src';
```
Comment (Wolftousen) [inline stories/Trends.stories.tsx]: see previous comment about index.ts referencing

File: stories/useGraph.tsx [hunk 11]
```diff
@@ -1,7 +1,6 @@
 import { useState } from 'react';
 import { getDateRange } from '../src/functions/dateFunctions';
-import { MetricThresholdSet } from '../src/interfaces/propInterfaces';
-import { DoraRecord } from '../src/interfaces/apiInterfaces';
+import { MetricThresholdSet, DoraRecord } from '../src';
```
Comment (Wolftousen) [inline stories/useGraph.tsx]: see previous comment about index.ts referencing

**Review by Wolftousen — DISMISSED** [review]

**Review by tjvkroll — DISMISSED** [review]

**Review by tjvkroll — APPROVED** [review]

**Pull request merged**